bin/ineffassign

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -eu
+
+cd "$(pwd -P)"
+
+bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+rootdir="$( cd $bindir/.. && pwd )"
+
+cd "$rootdir"
+
+# install ineffassign from vendor
+go install ./vendor/github.com/gordonklaus/ineffassign
+
+ineffassign ./
```

Comment:
TIOLI This feels really similar to linting to me. How would you feel about updating the `go/lint` script to also run this tool, instead of adding a separate script? I can imagine that in almost all cases it would make sense to run them together.

Reply:
Good point. I started down this path, then thought about what other tools we could combine here. I landed on [golangci-lint](https://github.com/golangci/golangci-lint/), which provides golint, ineffassign, govet, and a bunch of other linters, and it's fast. It will also let us incrementally enable linters as we go. Let me know what you think.

Reply:
Awesome! Yeah, using golangci-lint works for me. In fact, @radu-matei suggested that a while ago, here: https://github.com/linkerd/linkerd2/pull/2018#discussion_r243457453. I'm all for switching to that setup.